### ISSUE TITLE
Snippets with hard tabs fail to parse

### ISSUE THREAD

**Issue by victorxw** [atom/snippets#101]
```coffeescript
'.source.c':
	'main function':
		'prefix': 'mainf'
		'body': """
	int main(int argc, char const *argv[]) {
		${1:/*code here*/ }
		return 0;
	}
	"""
```
I get this error message:
> Failed to load snippets from '/Users/xxx/.atom/snippets.cson'
Syntax error on line 4, column 11: Unexpected token

Can anyone provide a fix?

**Comment by 50Wliu** [comment]
Try using \n and \t instead of actual newlines.

**Comment by victorxw** [comment]
Thanks for the reply, but that's only a temporary fix. I hope this issue gets resolved in future updates of atom.

**Comment by lee-dohm** [comment]
I pasted your snippet into my `snippets.cson` and did not get the error message you show. Could you give the following information?

* Which version of Atom are you using?
* What OS and version are you running?
* Do you have the [Shadow DOM feature](http://blog.atom.io/2015/01/05/enable-shadow-dom.html) enabled?
* Can you reproduce the issue after completely exiting Atom and restarting using `atom --safe`?

**Comment by victorxw** [comment]
Sure details as following:

* atom version: 0.175.0
* OS: Mac OS X 10.10.1 Build14B25
* Shadow DOM feature: Enabled
* Issue persists after restart atom using `atom --safe`

and issue shows no change even if I disable shadow DOM feature and restart Atom.

**Comment by filipminev** [comment]
I'm having the same issue as @victorxw. @50Wliu answer fixes the error, but it's unhandy.

**Comment by lee-dohm** [comment]
I posted my `snippets.cson` as a Gist here:

https://gist.github.com/lee-dohm/acdefbae191a92e23577

Perhaps you can see if there is a difference in how mine is formatted from yours?

**Comment by filipminev** [comment]
The issue seems to be only if you are using tabs instead of spaces. If I use your ```main function``` example and replace the spaces with tabs, it gives me the ```Unexpected token``` error.

**Comment by lee-dohm** [comment]
@kevinsawicki Could this have been caused by the migration to the new CSON library?

**Comment by kevinsawicki** [comment]
> Could this have been caused by the migration to the new CSON library?

Definitely, will take a look.

**Comment by kevinsawicki** [comment]
Looks like a hard tabs bug in the CSON parsing we are using, https://github.com/groupon/cson-safe/issues/20

**Comment by kevinsawicki** [comment]
This will be fixed in the next Atom release, 0.177

**Comment by lee-dohm** [comment]
Thanks @kevinsawicki!
